### ISSUE TITLE
How to deal with multiple retrieval tasks?

### ISSUE THREAD

**Issue by hamdi1611** [tensorflow/recommenders#139]
in the [multi-task tutorial](https://www.tensorflow.org/recommenders/examples/multitask), it is said that "We can expect better results when we can transfer knowledge from a data-abundant task (such as clicks) to a closely related data-sparse task (such as purchases)". So, I assume that there will be two retrieval tasks on different type of data.

For example, if I have transaction data with columns (type, user, item):
click, user A, item X
click, user A, item Y
click, user A, item Z
purchase, user A, item Z
click, user B, item X
click, user B, item Y
purchase, user B, item X
etc

How to separate these two tasks (view and purchase)?

and how to set the weights for each task?
for example, final_loss = 1 * purchase_retrieval_loss + 0.1 * view_retrieval_loss

**Comment by maciejkula** [comment]
On the first question: I'm not quite sure what you mean. There are many ways to separate tensor/dataset like this into multiple inputs; you can also use example weights to compute losses on a subset of elements in your tensors.

On the second question: the weights to combining the losses are an important hyperparameter to tune. Make sure you monitor the accuracy on both tasks to find a good operating point!

**Comment by hamdi1611** [comment]
This is what I found in tutorial when there is a retrieval task and a ranking task:
`rating_loss = self.rating_task(
        labels=ratings,
        predictions=rating_predictions,
    )`
    `retrieval_loss = self.retrieval_task(user_embeddings, movie_embeddings)`

So, what should I do if there are two retrieval tasks since each task should handle its own data?
`#user_embeddings and item_embeddings may refer to purchase xor view type of data ??`
`purchase_retrieval_loss = self.retrieval_task(user_embeddings ??, item_embeddings ??)`
`view_retrieval_loss = self.retrieval_task(user_embeddings ??, item_embeddings ??)`

**Comment by maciejkula** [comment]
There are a couple of ways of handling this. 

1. You could select the rows of `user_embeddings` and `movie_embeddings` that correspond to each task and pass these subsets into the tasks.
2. You could create sample weight tensors that are 1 if a row corresponds to the current task and 0 otherwise, and use them as the `sample_weight` argument.

**Comment by hamdi1611** [comment]
Got it. Thank you!
